File: basics/string-comparison/clojure/StringComparison.clj
(def s1 (nth *command-line-args* 0))
(def s2 (nth *command-line-args* 1))

(if (= s1 s2) (println "'" s1 "'" "is equal to" "'" s2 "'" "with case."))
(if (= (clojure.string/lower-case s1) (clojure.string/lower-case s2)) (println "'" s1 "'" "is equal to"  "'" s2 "'" "without case."))

(if (< (compare s1 s2) 0) (println "'" s1 "'" "is lexically before"  "'" s2 "'" "."))
(if (= (compare s1 s2) 0) (println "'" s1 "'" "is lexically equal"  "'" s2 "'" "."))
(if (> (compare s1 s2) 0) (println "'" s1 "'" "is lexically after"  "'" s2 "'" "."))

File: basics/string-comparison/rust/StringComparison.rs
use std::cmp::Ordering;
use std::env;

fn main() {
    let mut args: Vec<_> = env::args().skip(1).collect();
    // Since we are popping the stack, arg2 comes first
    let s2 = args.pop().unwrap();
    let s1 = args.pop().unwrap();

    // These 2 comparison methods are equivalent
    if s1 < s2 {
        println!("{:?} is lexically before {:?}.", s1, s2);
    }
    if s1 == s2 {
        println!("{:?} is lexically equal to {:?}.", s1, s2);
    }
    if s1 > s2 {
        println!("{:?} is lexically after {:?}.", s1, s2);
    }

    match s1.cmp(&s2) {
        Ordering::Less    => println!("{:?} is lexically before {:?}.", s1, s2),
        Ordering::Equal   => println!("{:?} is lexically equal to {:?}.", s1, s2),
        Ordering::Greater => println!("{:?} is lexically after {:?}.", s1, s2),
    }
}
